upgrade_test.py

```diff
@@ -159,6 +159,12 @@ def upgrade_node(self, node, upgrade_sstables=True):
         new_scylla_repo = self.params.get('new_scylla_repo')
         new_version = self.params.get('new_version')
         upgrade_node_packages = self.params.get('upgrade_node_packages')
+        scylla_yaml_updates = {
```

Comment:
when doing rollbacks, shouldn't we recover settings?

Reply:
We kinda do, at least on reinstalls: https://github.com/scylladb/scylla-cluster-tests/blob/1d3514749a4dd9679a291c02351ba76bfbbbf7bf/upgrade_test.py#L229

That being said, we don't reuse `scylla.yaml` as it was initially, but append feature flags to it when necessary. I already removed one update for the `enable_3_1_0_compatibility_mode` flag with this PR, but I don't see us avoiding these changes to `scylla.yaml`.